### PR TITLE
Support setting and getting a decription

### PR DESCRIPTION
**Pull request by tyll**
This contains commits from #57

### CODE REVIEW

File: tests/lib/nm/user_test.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+from libnmstate import nm
+
+
+@pytest.fixture
+def NM_mock():
+    with mock.patch.object(nm.user.nmclient, 'NM') as m:
+        yield m
+
+
+def test_create_setting_None(NM_mock):
```
Comment (EdDev) [inline tests/lib/nm/user_test.py]: Naming of tests suggestion:
- test_create_setting_with_no_values
- test_create_setting_given_existing_config
- test_create_setting_given_non_existing_config

**Comment by EdDev** [comment]
Two other comments:
- The commit subject: Can you please add 'interface'?  Description may appear in other places.
- Question: The test coverage decreased although you added tests, any idea why?

File: libnmstate/nm/user.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+#
+""" Use a Network Manager User Setting to store and retrieve a device
+description:
+
+https://lazka.github.io/pgi-docs/#NM-1.0/classes/SettingUser.html
+"""
+
+from libnmstate.nm import nmclient
+
+NMSTATE_DESCRIPTION = "nmstate.description"
```
Comment (EdDev) [inline libnmstate/nm/user.py]: This is interface specific, we may need other descriptions in the future.

File: tests/lib/nm/user_test.py (hunk 1 above)
Comment (tyll) [reply tests/lib/nm/user_test.py]: It tests the case where no settings needs to be created, how about `test_create_no_setting`?

File: libnmstate/nm/user.py (hunk 2 above)
Comment (tyll) [reply libnmstate/nm/user.py]: changed to `nmstate.interface.desciption`

File: tests/lib/nm/user_test.py (hunk 1 above)
Comment (EdDev) [reply tests/lib/nm/user_test.py]: sounds good, thanks.

**Comment by tyll** [comment]
Change the commit message. The tests do not cover everything and the coverage is less than the coverage of the other code I guess. A problem I see with the tests is that they need so much mocking that I wonder if we actually test it properly instead of just how we assume that NM works. Ideally we would mock NM in a way that it assumes a certain network configuration and the API works as designed instead of mocking the API hoping that the resembles the API properly.

**Comment by EdDev** [comment]
I agree, the over mocking needed makes these tests hard to write and even harder to maintain.
We could decide not to test these low level driver modules, reasoning that they do not contain any significant business logic and the burden of maintaining them is too high.
I think we can take this decision when we will cover them using the integration/functional tests.

**Pull request merged**